src/main/plugins/org.dita.base/xsl/preprocess/conrefImpl.xsl

```diff
@@ -559,10 +559,10 @@ See the accompanying LICENSE file for applicable license.
     <xsl:variable name="conref-gen-id" as="xs:string">
       <xsl:choose>
         <xsl:when test="empty($elemid) or $elemid = $href-elemid">
-          <xsl:value-of select="generate-id(key('id', $conref-topicid)[contains(@class, ' topic/topic ')]//*[@id = $href-elemid])"/>
+          <xsl:value-of select="generate-id((key('id', $conref-topicid)[contains(@class, ' topic/topic ')]//*[@id = $href-elemid])[1])"/>
```

Comment:
When `$conref-topicid` starts to return correct values, `id` key will also start returning values. XSLT 1.0 will accept multiple argument values for `generate-id()`, but XSLT 2.0 will not. Thus we have to explicitly use only the first.